File: node/sdk/controllers/publishController.js
import { DaprClient } from "@dapr/dapr";
import express from "express";
import { v4 as uuidv4 } from 'uuid';

const router = express.Router();

export const publishController = router.get("/", async (_req, res) => {
    try {
        const client = new DaprClient();
        const pubSubName = "pubsub";
        const topic = "orders";

        const orderId = uuidv4()
        const data = [
            { "orderId": orderId }
        ]

        await client.pubsub.publish(pubSubName, topic, data);

        res.json({ "msg": `Order published with order id: ${orderId}` })
    } catch (error) {
        console.error(error)
        res.json({ "err": error })
    }
})

File: go/sdk/controllers/publishController.go
package controllers

import (
	"bytes"
	"context"
	"encoding/json"
	"log"

	dapr "github.com/dapr/go-sdk/client"
	"github.com/gofiber/fiber/v2"
	"github.com/google/uuid"
)

type Order struct {
	OrderId string `json:"orderId"`
}

func PublishController(cxt *fiber.Ctx) error {
	uuid := uuid.New()

	orders := []Order{
		{
			OrderId: uuid.String(),
		},
	}
	// Marshal the orders struct to JSON
	t, err := json.Marshal(orders)
	client, err := dapr.NewClient()

	if err != nil {
		log.Fatal(err)
	}

	ctx := context.Background()

	if err := client.PublishEvent(ctx, "pubsub", "orders", bytes.NewBuffer(t)); err != nil {
		panic(err)
	}

	return cxt.JSON(fiber.Map{"msg": "Order published with order id: " + uuid.String()})
}
